### PR TITLE
fixed Deserialization exception on WaveScanRecord

### PR DESCRIPTION
**Pull request by munishchouhan**
this PR will fix Deserialization exception on WaveScanRecord

### CODE REVIEW

**Comment by munishchouhan** [comment]
tested for invalid:
Screenshot 2024-10-09 at 15 08 15

**Comment by munishchouhan** [comment]
tested for valid:
Screenshot 2024-10-09 at 15 08 03

File: src/main/groovy/io/seqera/wave/service/persistence/impl/SurrealPersistenceService.groovy [hunk 1]
```diff
@@ -246,7 +246,7 @@ class SurrealPersistenceService implements PersistenceService {
     WaveScanRecord loadScanRecord(String scanId) {
         if( !scanId )
             throw new IllegalArgumentException("Missing 'scanId' argument")
-        final statement = "SELECT * FROM wave_scan:$scanId FETCH vulnerabilities"
+        final statement = "SELECT * FROM wave_scan where id == 'wave_scan:⟨$scanId⟩' FETCH vulnerabilities"
```
Comment (pditommaso) [inline src/main/groovy/io/seqera/wave/service/persistence/impl/SurrealPersistenceService.groovy:249]: what those magic brackets do?

Comment (munishchouhan) [reply src/main/groovy/io/seqera/wave/service/persistence/impl/SurrealPersistenceService.groovy:249]: thats how the id stored in surrealdb `id: wave_scan:⟨sc-8c94bed81c2a999d_1⟩`

**Pull request merged**